pkg/service/providerindex/legacy.go

```diff
@@ -48,11 +75,20 @@ func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Mult
 	for _, claimCid := range claimsCids {
 		claim, err := ls.claimsStore.Get(ctx, cidlink.Link{Cid: claimCid})
 		if err != nil {
+			if errors.Is(err, types.ErrKeyNotFound) {
+				continue
+			}
+
 			return nil, err
 		}
 
-		pr, err := synthetizeProviderResult(claim)
+		pr, err := ls.synthetizeProviderResult(claim)
 		if err != nil {
+			var unsupportedClaimErr errUnsupportedClaimType
+			if errors.As(err, &unsupportedClaimErr) {
+				continue
+			}
+
 			return nil, err
```

Comment:
I would perhaps just always continue on error here (log it though), and not let a malformed capability cause no results to be returned.